cli/parse_pdfs.py

```diff
@@ -185,16 +190,16 @@ def download_pdf(
         return output_path
 
 
-def select_page_at_random(num_pages: int) -> bool:
+def select_page_at_random(num_pages: int, rng: float) -> bool:
     """Determine whether to include a page using a random number generator. Used for debugging.
 
     Args:
         num_pages: The number of pages in the PDF.
+        rng: A random number between 0 and 1.
 
     Returns:
         The page number to include.
     """
-    rng = np.random.random()
```

Comment:
How come we moved this outside the function, from what I can see it's only used here? 

Assuming you're trying to not pick the same page twice?

Reply:
Moved rng creation outside the function so that I could pass in an indexed rng from a loop. This allows us to set a seed value so that we get the same pages on debug runs (with png outputs plotted) to give better observability on the impact of changes.